### PR TITLE
Onboarding testnet 3 and error handling 

### PR DESCRIPTION
**Pull request by rosepuppy**
---

## Views

* `src/views/TransferStatus.tsx`
  * Render the error if discovered in status

* `src/views/forms/AccountManagementForms/DepositForm.tsx`
  * Use `parseWalletError` to handle wallet rejections

## Hooks

* `src/hooks/useLocalNotifications.tsx`
  *  catch the error that is thrown by the squid status and set it for the notification

* `src/hooks/useLocalNotifications.tsx`
  * Display the error if it is in status in the description
  * Add description to the customMenuContent

## Packages

* `@dydxprotocol/v4-client-js`
  * updated: 0.35.0 -> 0.36.1
 
* `@dydxprotocol/v4-localization`
  * updated: 0.1.5 -> 0.1.8

### CODE REVIEW

File: src/hooks/useLocalNotifications.tsx [hunk 1]
```diff
@@ -66,11 +66,14 @@ const useLocalNotificationsContext = () => {
...
         try {
           if (currentStatus && currentStatus?.squidTransactionStatus !== 'ongoing') continue;
-  
+
           const status = await squid?.getStatus({ transactionId: txHash, toChainId, fromChainId });
           if (status) statuses[txHash] = status;
         } catch (error) {
-          console.error(error);
+          // ignore not found errors since the route might not be available yet
+          if (error?.errors?.length && error.errors[0].errorType !== 'NotFoundError') {
```
Comment (jaredvu) [inline src/hooks/useLocalNotifications.tsx:74]: If you think that we will be handling other error types separately we should add the `NotFoundError` to a SquidErrors enum

File: src/hooks/useNotificationTypes.tsx [hunk 2]
```diff
@@ -119,27 +122,46 @@ export const notificationTypes = [
...
           const finished = Boolean(status) && status?.squidTransactionStatus !== 'ongoing';
           const type = toChainId === TESTNET_CHAIN_ID ? 'deposit' : 'withdraw';
+          // @ts-ignore status.errors is not in the type definition but can be returned
+          const error = status?.errors?.length ? status?.errors[0] : status?.error;
+
+          // TODO: confirm with design what the description should be
+          const description = (
+            <div>
+              <Styled.TransferText>
+                {type === 'deposit' ? 'Deposit of ' : 'Withdraw of'}{' '}
```
Comment (jaredvu) [inline src/hooks/useNotificationTypes.tsx:132]: `Deposit of ` (has an extra space) & `Withdraw of` should be added to v4-localization

File: src/hooks/useNotificationTypes.tsx [hunk 3]
```diff
@@ -119,27 +122,46 @@ export const notificationTypes = [
...
+                <Output type={OutputType.Fiat} value={toAmount} />
+              </Styled.TransferText>
+
+              {error && (
+                <AlertMessage type={AlertType.Error}>
+                  {stringGetter({
+                    key: STRING_KEYS.SOMETHING_WENT_WRONG_WITH_MESSAGE,
```
Comment (jaredvu) [inline src/hooks/useNotificationTypes.tsx:139]: `SOMETHING_WENT_WRONG_WITH_MESSAGE` is expecting a message param

File: src/views/TransferStatusSteps.tsx [hunk 4]
```diff
@@ -59,7 +59,7 @@ export const TransferStatusSteps = ({ status }: ElementProps) => {
       },
     ];
 
-    const currentStatus = routeStatus[routeStatus?.length - 1];
+    const currentStatus = routeStatus ? routeStatus[routeStatus?.length - 1] : undefined;
```
Comment (jaredvu) [inline src/views/TransferStatusSteps.tsx:62]:
```suggestion
    const currentStatus = routeStatus?.[routeStatus?.length - 1];
```

File: src/constants/wallets.ts [hunk 5]
```diff
@@ -341,7 +341,7 @@ export type DydxAddress = `dydx${string}`;
 export const DYDX_CHAIN_INFO: Parameters<typeof suggestChain>[0] = {
   rpc: '13.59.4.93:26657',
   rest: '13.59.4.93:1317',
-  chainId: 'dydx-testnet-2',
+  chainId: 'dydx-testnet-3',
```
Comment (jaredvu) [inline src/constants/wallets.ts:344]: does the rpc/rest ip match with this chainId?

Comment (rosepuppy) [reply src/constants/wallets.ts:344]: oops, good call

**Review by jaredvu — APPROVED** [review]

**Pull request merged**